ui/static/ui/js/listing/import_status.jsx

```diff
@@ -0,0 +1,130 @@
+define("import_status", ["React", "lodash", "jquery", "utils",
+  "react_spinner", "status_box"],
+  function (React, _, $, Utils, ReactSpinner, StatusBox) {
+    'use strict';
+
+    return React.createClass({
+      getInitialState: function () {
+        return {
+          importTasks: {},
+          hasRefreshed: false
+        };
+      },
+      updateImportStatus: function () {
+        var thiz = this;
+
+        Utils.getCollection("/api/v1/tasks/").then(function (tasks) {
+          // Only deal with tasks on this repo that deal with imports.
+
+          tasks = _.filter(tasks, function(task) {
+            return (task.task_type === 'import_course' &&
+              task.task_info.repo_slug === thiz.props.repoSlug);
+          });
+          var tasksMap = {};
+          _.each(tasks, function (task) {
+            tasksMap[task.id] = task;
+          });
+          thiz.setState({importTasks: tasksMap});
+
+          var notProcessing = _.filter(tasks, function (task) {
+            return task.status === 'success' || task.status === 'failure';
+          });
+
+          var deferred = $.Deferred();
+
+          // Delete one after another to avoid race conditions with
+          // Django session.
+          _.each(notProcessing, function (task) {
+            deferred.then(function () {
+              return $.ajax({
+                method: 'DELETE',
+                url: '/api/v1/tasks/' + task.id + '/'
+              });
+            });
+          });
+          deferred.resolve();
```

Comment:
The code above attaches handlers for the success case of `deferred` but doesn't actually resolve or reject the initial promise until this line. We need to resolve the initial promise so that it can start evaluating the other success callbacks. To illustrate:

    deferred -> task1 -> task2 -> task3

`deferred` is always resolved. It doesn't do anything but we need to attach the task chain to something we have control of. Then `task1` is evaluted. It returns a promise that's either rejected or resolved depending on how the AJAX request went. If `task1` succeeded `task2`'s success handler is then executed.

If `task1` failed it would try to execute `task2`'s failure handler, or `task3`, or whatever the next in line with a failure handler is. None of them have failure handlers so we just stop executing the chain in the event of a AJAX failure.